### PR TITLE
Add provider-neutral secret store support

### LINKED ISSUE #59: Secret Store Abstraction
Provide a provider-neutral secret store for sensitive values (API keys, signing keys, encryption secrets) that today live in Fastly SecretStore.

See [ROADMAP.md](https://github.com/stackpop/edgezero/blob/main/ROADMAP.md#epic-3-secret-store-abstraction)

### PR DESCRIPTION
**Pull request by prk-Jr**
## Summary

- Add a provider-neutral runtime secret store so handlers can read secrets consistently across Fastly, Cloudflare, and Axum.
- Wire manifest-driven `[stores.secrets]` configuration into each adapter while preserving the existing low-level manual dispatch behavior.
- Add security hardening, demo coverage, smoke-test support, and docs so the new secret-store path is safer to adopt and review.

## Design

`SecretStore` is a single two-argument trait — `get_bytes(store_name, key)` — mirroring the `KvStore` / `KvHandle` naming convention. `SecretHandle` is the ergonomic cloneable wrapper injected into request context. Platforms with named secret stores (Fastly) use `store_name` to open the right store; Cloudflare and Axum ignore it (flat namespaces).

## Changes

| Crate / File | Change |
| ------------ | ------ |
| `crates/edgezero-core` | Added two-arg `SecretStore` trait (`get_bytes(store_name, key)`), `SecretHandle` wrapper, sanitized `SecretError` mapping, request-context/extractor support, `InMemorySecretStore` + `NoopSecretStore` test helpers, `secret_store_contract_tests!` macro, and `[stores.secrets]` manifest schema with adapter overrides. |
| `crates/edgezero-adapter-fastly` | Added `FastlySecretStore` (opens a named Fastly SecretStore per lookup), manifest-aware secret wiring, and a private shared store-handle injection path for KV + secrets. |
| `crates/edgezero-adapter-cloudflare` | Added `CloudflareSecretStore` (ignores store name; flat namespace), manifest-aware secret wiring, and a private shared store-handle injection path for KV + secrets. |
| `crates/edgezero-adapter-axum` | Added `EnvSecretStore` (reads from env vars; ignores store name), service/dev-server secret injection, and TCP integration tests for secret-present / missing / no-store cases. |
| `crates/edgezero-cli/src/main.rs` | Added adapter-specific secret-store binding messages during `edgezero build`. |
| `examples/app-demo/crates/app-demo-core/src/handlers.rs` | Added the `secrets_echo` demo handler with allowlisted smoke-test keys and tests for success, sanitized failure, and invalid input. |
| `examples/app-demo/edgezero.toml` | Added the demo route and `[stores.secrets]` configuration. |
| `examples/app-demo/crates/app-demo-adapter-fastly/fastly.toml` | Added local secret-store config for Fastly smoke testing. |
| `scripts/smoke_test_secrets.sh` | Added adapter smoke checks for Axum, Fastly, and Cloudflare secret access. |
| `docs/guide/configuration.md` | Documented runtime secret-store configuration and the distinction from `[[environment.secrets]]`. |

## Closes

Closes #59

## Test plan

- [x] `cargo test --workspace --all-targets`
- [x] `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- [x] `cargo check --workspace --all-targets --features "fastly cloudflare"`
- [ ] WASM builds: `wasm32-wasip1` (Fastly) / `wasm32-unknown-unknown` (Cloudflare)
- [ ] Manual testing via `edgezero-cli dev`
- [x] Other: `cargo test -p app-demo-core`, `bash -n scripts/smoke_test_secrets.sh`, `./docs/node_modules/.bin/prettier --check docs/guide/configuration.md`, `npm --prefix docs run build`

## Checklist

- [x] Changes follow [CLAUDE.md](/CLAUDE.md) conventions
- [x] No Tokio deps added to core or adapter crates
- [x] Route params use `{id}` syntax (not `:id`)
- [x] Types imported from `edgezero_core` (not `http` crate)
- [x] New code has tests
- [x] No secrets or credentials committed

### CODE REVIEW

**Review by ChristianPavilonis — COMMENTED** [review]
## PR Review: Add provider-neutral secret store support

**Overall**: Well-structured implementation that cleanly mirrors the KV store pattern. Security hardening is thoughtful (sanitized errors, name validation, demo handler allowlisting). Comprehensive test coverage across all adapters. A few API design and code duplication issues worth addressing.

**CI**: All checks passing.

**Findings**: 3 High, 7 Medium, 4 Low (details in inline comments below)

### Positives
- Excellent security hardening — secret names sanitized in error conversion, verified by tests
- Clean architectural parity with KV store pattern
- `validate_name` rejects empty, control-char, and oversized names
- Demo handler allowlist prevents secret exfiltration
- Comprehensive test coverage (460+ lines, contract tests, TCP integration tests)
- Proper `#[must_use]` on builder methods

File: crates/edgezero-adapter-axum/src/service.rs [hunk 1]
```diff
@@ -142,6 +161,48 @@ mod tests {
...
     }
 
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn with_secret_handle_injects_into_request() {
+        use crate::secret_store::EnvSecretStore;
+        use edgezero_core::secret_store::SecretHandle;
+        use std::sync::Arc;
+
+        let _guard = env_guard().lock().await;
+        let _env = EnvOverride::set("__EDGEZERO_SERVICE_TEST_SECRET__", "injected_value");
```
Comment (aram356) [inline crates/edgezero-adapter-axum/src/service.rs]: 🤔 **`set_var` in multi-thread test**: This test uses `flavor = "multi_thread", worker_threads = 2` while calling `set_var` via `EnvOverride`. Technically unsound on Rust 1.66+ (data race if Tokio workers read env concurrently). The mutex serialization mitigates practical risk, but consider refactoring to use `InMemorySecretStore` here instead of `EnvSecretStore` to avoid the issue entirely.

File: crates/edgezero-adapter-cloudflare/src/secret_store.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    /// Create a secret store from a cloned `Env`.
+    pub fn from_env(env: worker::Env) -> Self {
+        Self { env }
+    }
+}
+
+#[cfg(all(feature = "cloudflare", target_arch = "wasm32"))]
+#[async_trait(?Send)]
+impl SecretStore for CloudflareSecretStore {
+    async fn get_bytes(&self, _store_name: &str, key: &str) -> Result<Option<Bytes>, SecretError> {
```
Comment (aram356) [inline crates/edgezero-adapter-cloudflare/src/secret_store.rs:41]: 🤔 **Flat namespace silently ignores `store_name`**: Correct for Cloudflare's platform model, but a user expecting multi-store isolation will get surprising behavior. Future enhancement: consider logging a warning when `store_name` doesn't match the configured store name.

File: crates/edgezero-core/src/secret_store.rs [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+    /// A validation error (e.g., invalid secret name).
+    #[error("validation error: {0}")]
+    Validation(String),
+
+    /// A general internal error.
+    #[error("secret store error: {0}")]
+    Internal(#[from] anyhow::Error),
+}
+
+impl From<SecretError> for EdgeError {
```
Comment (aram356) [inline crates/edgezero-core/src/secret_store.rs:53]: 😃 **Excellent security discipline**: The `SecretError → EdgeError` conversion deliberately strips secret names and internal details from user-visible error messages, with an explicit test verifying no leakage. Thoughtful security-by-default design.

**Review by aram356 — APPROVED** [review]
## PR Review

### Summary

Well-designed provider-neutral secret store that follows the established KvStore/KvHandle pattern exactly. Strong test coverage (477 passing), good security discipline on error sanitization, and clean manifest integration with per-adapter overrides. Ready to merge.

### 😃 Praise

- Excellent security-by-default: `SecretError → EdgeError` strips secret identifiers from user-visible errors, verified by explicit tests
- Clean adapter pattern consistency: core trait, Arc-wrapped handle, FromRequest extractor, contract test macro — mirrors KvStore perfectly
- Thorough test coverage including validation edge cases, security leak tests, contract tests, and TCP integration tests
- Well-structured `[stores.secrets]` manifest schema with global defaults and per-adapter overrides

### Findings

#### Non-blocking

- 🤔 **`set_var` in multi-thread test context**: `service.rs` test uses `flavor = "multi_thread"` with `EnvOverride`; consider using `InMemorySecretStore` instead to avoid the `set_var` unsoundness in a follow-up
- 🤔 **Flat-namespace adapters silently ignore `store_name`**: Correct for Cloudflare/Axum, but could surprise users expecting multi-store isolation; future enhancement could log a warning
- ♻️ **Duplicate `store_handles.rs`**: Fastly and Cloudflare versions are nearly identical (~63 lines); could be extracted to shared utility in a future cleanup

### 📌 Out of Scope

- Spin adapter secret store (Spin has `spin_sdk::variables`) — worth tracking as a follow-up

### CI Status

- fmt: PASS
- clippy: PASS
- tests: PASS (477 tests)

File: crates/edgezero-adapter-cloudflare/src/secret_store.rs (hunk 2 above)
Comment (prk-Jr) [reply crates/edgezero-adapter-cloudflare/src/secret_store.rs:41]: Checked this. The ignore is intentional for Cloudflare because Workers secrets are flat-namespace and the configured store name is metadata only there. I’m leaving the warning as follow-up rather than widening this PR’s runtime behavior; the compare point only exists cleanly on the manifest-driven path, so I’d rather design that separately.
